core/src/subgraph/registrar.rs

```diff
@@ -518,18 +521,38 @@ async fn resolve_start_block(
         .expect("cannot identify minimum start block because there are no data sources")
     {
         0 => Ok(None),
-        min_start_block => chain
-            .block_pointer_from_number(logger, min_start_block - 1)
-            .await
-            .map(Some)
-            .map_err(move |_| {
-                SubgraphRegistrarError::ManifestValidationError(vec![
-                    SubgraphManifestValidationError::BlockNotFound(min_start_block.to_string()),
-                ])
-            }),
+        min_start_block => Retry::spawn(retry_strategy(Some(2)), move || {
+            chain
+                .block_pointer_from_number(&logger, min_start_block - 1)
+                .inspect_err(move |e| warn!(&logger, "Failed to get block number: {}", e))
+        })
+        .await
+        .map(Some)
+        .map_err(move |_| {
+            SubgraphRegistrarError::ManifestValidationError(vec![
+                SubgraphManifestValidationError::BlockNotFound(min_start_block.to_string()),
+            ])
+        }),
     }
 }
 
+fn retry_strategy(limit_opt: Option<usize>) -> Box<dyn Iterator<Item = Duration> + Send> {
+    // Exponential backoff, but with a maximum
+    let max_delay_ms = 30_000;
+    let backoff = ExponentialBackoff::from_millis(2)
+        .max_delay(Duration::from_millis(max_delay_ms))
+        .map(jitter);
+
+    // Apply limit (maximum retry count)
+    match limit_opt {
+        Some(limit) => {
+            // Items are delays *between* attempts,
+            // so subtract 1 from limit.
+            Box::new(backoff.take(limit - 1))
+        }
+        None => Box::new(backoff),
+    }
+}
```

Comment:
This looks identical to `graph::util::futures::retry_strategy`, correct? If so, we can make that public and reuse it here, and also avoid `pub use tokio_retry`.